docs: rewrite Kubernetes page (#3928)
docker-mailserver/docker-mailserver: docs/content/examples/tutorials/mailserver-behind-proxy.md

Some deployments may require a service to route traffic (kubernetes) when deploying, in which case the below advice is important to understand well.

> We also provide a [dedicated documentation page][docs::kubernetes] for using a proxy in Kubernetes.

Reviewer: ```suggestion
The guide here has also been adapted for [our Kubernetes docs][docs::kubernetes].
```

    You should adjust configuration of these monitoring services to monitor for auth failures from those services directly instead, adding an exclusion for that service IP from any DMS logs monitored (_but be mindful of PROXY header forgery risks_).

> [docs::kubernetes]: ../../../config/advanced/kubernetes/#__tabbed_2_4

Reviewer: Excess `../` + `.md` extension for a page is usually required AFAIK?:

```suggestion
[docs::kubernetes]: ../../config/advanced/kubernetes.md#__tabbed_2_4
```
